### PR TITLE
Added Triangle Shape

### PR DESCRIPTION
**Pull request by makermelissa**
Enjoy! It doesn't have stroke at this time, but this is an initial working triangle.

### CODE REVIEW

File: adafruit_display_shapes/triangle.py [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+            print("outline")
+            self._palette[1] = outline
+            self._line(x0 - smallest_x, 0, x1 - smallest_x, y1 - y0, 1)
+            self._line(x1 - smallest_x, y1 - y0, x2 - smallest_x, y2 - y0, 1)
+            self._line(x2 - smallest_x, y2 - y0, x0 - smallest_x, 0, 1)
+
+        super().__init__(self._bitmap, pixel_shader=self._palette, x=smallest_x, y=y0)
+
+    # pylint: disable=invalid-name, too-many-locals, too-many-branches
+    def _helper(self, x0, y0, x1, y1, x2, y2):
```
Comment (tannewt) [inline adafruit_display_shapes/triangle.py]: What does this do?

File: adafruit_display_shapes/triangle.py [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+            x1, x2 = x2, x1
+
+        if y0 > y1:
+            y0, y1 = y1, y0
+            x0, x1 = x1, x0
+
+        smallest_x = x0
+        largest_x = x0
+
+        # Find the largest and smallest X values to figure out width for bitmap
```
Comment (tannewt) [inline adafruit_display_shapes/triangle.py:76]: Please use min and max to simplify this logic:

```python
xs = [x0, x1, x2]
width = max(xs) - min(xs) + 1
```

**Review by tannewt — CHANGES_REQUESTED** [review]
Thanks for adding triangle! Just a suggestion to simplify the logic.

File: adafruit_display_shapes/triangle.py (hunk 1 above)
Comment (makermelissa) [reply adafruit_display_shapes/triangle.py]: It draws the filled triangle. I can rename.

File: adafruit_display_shapes/triangle.py (hunk 2 above)
Comment (makermelissa) [reply adafruit_display_shapes/triangle.py:76]: Nice! I just learned something :)

**Comment by makermelissa** [comment]
Changes made

**Review by tannewt — APPROVED** [review]
I'm ok as is but have one other suggestion if you agree. I'd rename `fill` and `outline` to `fill_color` and `outline_color` because they read more as actions (aka functions) now. Furthermore, the names will make it tougher to add other properties like `outline_width` later. Food for thought. :-)

**Comment by makermelissa** [comment]
I used `fill` and `outline` to remain consistent with the other shapes. I don't want to rename all of the shapes because that would most likely break a bunch of code. Also, `outline_width` is called `stroke` on the other shapes, so that shouldn't be a problem. :)

**Comment by makermelissa** [comment]
The only other change I was considering making (which can totally be done in a future release without breaking anything) is to add a Line shape and subclass that for the triangle to reduce code.

**Pull request merged**